x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/IndicesAccessControl.java

```diff
@@ -36,11 +38,19 @@ public class IndicesAccessControl {
     public static final IndicesAccessControl DENIED = new IndicesAccessControl(false, Collections.emptyMap());
 
     private final boolean granted;
-    private final Map<String, IndexAccessControl> indexPermissions;
+    private final Supplier<Map<String, IndexAccessControl>> indexPermissions;
```

Comment:
For clarity's sake, please rename `indexPermissions` to `indexPermissionsSupplier`, and change the field type to `CachedSupplier`. It helps with readability if the more concrete type is obvious.